torch/csrc/lazy/core/tensor_impl.cpp

```diff
@@ -162,7 +189,8 @@ void LTCTensorImpl::setup_size_properties() {
 at::IntArrayRef LTCTensorImpl::sizes() const {
```

Comment:
@ezyang we could just override ::sizes for LTC to use `upper_bounds` instead if we are storing sym sizes in sizes_and_strides_ . Alternatively, we could continue storing upper bounds in `sizes_and_strides_` and override `sym_sizes` on LTC to return that field. For AOTAutograd we would still store sym ints into `sizes_and_strides_`. It's a bit inconsistent but might be less disruptive for XLA/LTC

fyi: @JackCaoG  @miladm

Reply:
this should not affect user side of story right? When user query `tensor.size()` they will still see the upper bound/static size. In the internal code base we need to be careful about.

Reply:
If XLA/LTC is going to report upper bound when you say size(), then I think you should store upper bound in `sizes_and_strides_`. To get `sym_sizes()` going you would have *that* query the `sym_sizes_` field. I agree it's inconsistent but as I said, if we are too afraid to make the change in XLA/LTC right now, storing `sym_sizes_` out of line is the expedient choice. Luckily, we won't need the SymSizeList Python object because AOTAutograd will be doing it properly.

Reply:
@ezyang sooo inconsistent!!!!!!!! I liked `sym_sizes_ ` field option even less than this version, but I'll rely on your experience that the `sym_sizes_` version is more self-contained.

Reply:
It is very inconsistent. The Correct(TM) thing to do is drag XLA kicking and screaming into the full sizes world, so that size() doesn't report upper bound but the real thing (and a symbolic int if we don't know). But that's more work.